### ISSUE TITLE
Is there a actual legit trade confirmation fix?

### ISSUE THREAD

**Issue by frosty4532** [Jessecar96/SteamDesktopAuthenticator#868]
I don't want to sound like a broken record, but I don't want to trust these fixes others have put out when I have no knowledge in programming. Does anyone know a safe fix please.

**Comment by dyc3** [comment]
If you are interested, I maintain steamguard-cli. I just released an update that fixes login and confirmations. It's capable of reading the exactly the same maFiles that SDA does, so you don't have to deal with a trade ban. You can download it here: https://github.com/dyc3/steamguard-cli/releases

Obviously, I'm biased because I'm the one who made it. If you don't trust the binaries in the releases, you can easily build it yourself.

**Comment by frosty4532** [comment]
so is the main issue with jesiccas the new ui not working, like is there anyway i can run jessicas code and edit the code to accept the confirmation with code?

**Comment by dyc3** [comment]
_Technically_, yes. But if you have to ask that question you aren't going to be able to make those changes. If you feel comfortable with it, there's the builds linked in #859. But I don't trust them, I haven't personally reviewed it.

**Comment by frosty4532** [comment]
so is there anyway you can tell me how i can edit the code to ignore the ui but beablke to run something in vs code to confirm trades if i type 'y' for example or no

**Comment by dyc3** [comment]
No.

**Comment by Jianchitz** [comment]
@dyc3 it kinda scares me how you lately started posting the same thing under every issue "dont rely on sda, dont wait sda" and now "i maintain this". Im not trying to make accuastions but its just looking a bit scary.

**Comment by dyc3** [comment]
I have been posting and answering issues on steam desktop authenticator for years. I have contributed to this repo, I literally have the contributor badge. If you look at my profile, I am very active on my projects.

steamguard-cli has a couple hundred stars, and a commit history that spans back to 2016. It's relied upon by a significant number of people and that's something I don't take for granted. Whether or not you believe me is irrelevant. The facts are that I've spent a great deal of time working on steamguard-cli, and I care about whether or not it's functional. These facts are easily verifiable.

If you don't feel comfortable using a third party app, you should move your authenticator to the steam mobile app. That's where you are going to have the most recourse with steam support if something goes wrong.

**Comment by DOTBU** [comment]
> I have been posting and answering issues on steam desktop authenticator for years. I have contributed to this repo, I literally have the contributor badge. If you look at my profile, I am very active on my projects.
> 
> steamguard-cli has a couple hundred stars, and a commit history that spans back to 2016. It's relied upon by a significant number of people and that's something I don't take for granted. Whether or not you believe me is irrelevant. The facts are that I've spent a great deal of time working on steamguard-cli, and I care about whether or not it's functional. These facts are easily verifiable.
> 
> If you don't feel comfortable using a third party app, you should move your authenticator to the steam mobile app. That's where you are going to have the most recourse with steam support if something goes wrong.

Can you tell me how to use it i have windows

**Comment by dyc3** [comment]
Instructions for how steamguard-cli works are here: https://github.com/dyc3/steamguard-cli/blob/master/README.md

If this information isn't clear enough, feel free to ask questions here: https://github.com/dyc3/steamguard-cli/discussions/new?category=q-a

**Comment by marxzcryt** [comment]
> I have been posting and answering issues on steam desktop authenticator for years. I have contributed to this repo, I literally have the contributor badge. If you look at my profile, I am very active on my projects.
> 
> steamguard-cli has a couple hundred stars, and a commit history that spans back to 2016. It's relied upon by a significant number of people and that's something I don't take for granted. Whether or not you believe me is irrelevant. The facts are that I've spent a great deal of time working on steamguard-cli, and I care about whether or not it's functional. These facts are easily verifiable.
> 
> If you don't feel comfortable using a third party app, you should move your authenticator to the steam mobile app. That's where you are going to have the most recourse with steam support if something goes wrong.

can you make a guide how can we do SGcli or at least upload on your youtube it would be good

**Comment by papichulooo28** [comment]
@dyc3 have you reviewed the solution posted on #859 ? if it is safe or something Cause I have used it, and I find you more reliable than the other guy. lol I think im in trouble lmao. Hoping you could personally check them so others could be reminded whether it is safe or not. We would really appreciate your help.

**Comment by b0Ni** [comment]
u can use archi 2fa

**Comment by papichulooo28** [comment]
@b0Ni but its too late I already used the "temporary fix" one and if its shady and he already got into my sda thingy and could get my items stolen after 7 days, so before that happens I just want to ask @dyc3 his opinion as to the legitimacy of the supposed fix while its not too late to create a new account and transfer all my shit.

**Comment by aavvaavvaa** [comment]
> I don't want to sound like a broken record, but I don't want to trust these fixes others have put out when I have no knowledge in programming. Does anyone know a safe fix please.

If you still scared to use new SDA or steamguard-cli, you can do your own code, this is a clearest way to be sure that your account and skins are safe
https://github.com/Jessecar96/SteamDesktopAuthenticator/issues/854#issuecomment-1604216514
https://github.com/Jessecar96/SteamDesktopAuthenticator/issues/854#issuecomment-1604659951
https://github.com/Jessecar96/SteamDesktopAuthenticator/issues/854#issuecomment-1605311661

**Comment by dyc3** [comment]
@papichulooo28 No I haven't reviewed it. I can't really speak to it's legitimacy.

**Comment by dennis0555** [comment]
> Instructions for how steamguard-cli works are here: https://github.com/dyc3/steamguard-cli/blob/master/README.md
> 
> If this information isn't clear enough, feel free to ask questions here: https://github.com/dyc3/steamguard-cli/discussions/new?category=q-a

can it also read encrypted mafifles?

**Comment by dyc3** [comment]
Yes

**Comment by MakcStudio** [comment]
A fully functional version of the SDA. Works and confirmation of trades and adding accounts. You are welcome to use it, I'm not a rookie developer, I work officially on big projects

https://github.com/MakcStudio/SteamDesktopAuthenticator

**Comment by Miracle7z** [comment]
Dont use MakcStudio. 

![1](https://github.com/Jessecar96/SteamDesktopAuthenticator/assets/137804715/c502780c-976e-45a0-be3b-574c85ba01a4)

The program is a trojan.
He won't steal the accounts/skins now. He'll let as many people trust the project as possible, and when he thinks he needs it, he'll steal them all at once.

**Comment by papichulooo28** [comment]
@Miracle7z have you checked the same thing with the other one?

**Comment by Miracle7z** [comment]
> have you checked the same thing with the other one?

Only Makc's program was barred. The original file does not.
He won't steal your account now, because by doing so, no one else will download the program. He will wait for as many people to download to apply the scam.

**Comment by MakcStudio** [comment]
It is open source, smartass. Do you think I have nothing to do and I first fixed the sda and then sewed the virus into it? You are so annoying to me along with the second fool in my issue, you write complete nonsense. If you knew any better, you wouldn't be here. If I wanted to steal something, believe me your antivirus would consider my program as my best friend.

My project is open source, just open it and look. And if you do not understand, and you just put the files on the virrostotal, without any understanding of how it works in general, do not write nonsense.

**Comment by jorgermduarte** [comment]
@MakcStudio Your fix does not make any sense, when it starts changing 30+ files.
The main problem is at the Steam Auth lib, and why do you keep promoting your own repo instead of a clean working pull request?

im tired of this kinda bullshit of you trying to redirect users to your own repo with doubtful code.

I created a pull request in the SDA with a clean fix changing only some files.
The main problem is on the SteamAuth lib since the endpoints from Steam had changed.

For anyone who wants to know:
https://github.com/Jessecar96/SteamDesktopAuthenticator/pull/882

Given this fix, anyone can build is own SDA.
I don't recommend downloading random .exe's from people you don't know.

**Comment by allendaydayup** [comment]
> @MakcStudio Your fix does not make any sense, when it starts changing 30+ files. The main problem is at the Steam Auth lib, and why do you keep promoting your own repo instead of a clean working pull request?
> 
> im tired of this kinda bullshit of you trying to redirect users to your own repo with doubtful code.
> 
> I created a pull request in the SDA with a clean fix changing only some files. The main problem is on the SteamAuth lib since the endpoints from Steam had changed.
> 
> For anyone who wants to know: #882
> 
> Given this fix, anyone can build is own SDA. I don't recommend downloading random .exe's from people you don't know.

译文：

I will continue to pay attention to your updates to the sda project. I originally compiled and used the source code myself, and I am trying to compile and use the code you submitted, which is very good

**Comment by MakcStudio** [comment]
In your fix, can you link new accounts to the sda? In mine you can. Your whole fix is a modified link in steamauth? You are super good, you did a great job!)

You can read my repo and find out why I didn't do a fork (at least read the first post) . This will give you the answer to who I am and why I can be trusted

**Comment by jorgermduarte** [comment]
No you cant be trusted mate :) you can say you are God, you can't be trusted when you have an account created on May 2023 and trying to redirect users out of SDA to use your doubtful .exe's releases.

Yes i didn't changed for the new accounts yet :) but people with maFiles already can setup the bots.  the fix for the new accounts probably will be just more a few minutes :) and will not take 30+ files like yours.

Mate you must be drunk

**Comment by jorgermduarte** [comment]
Well, I'm done with the conversation since your not worth of my time.
Everyone knows by this time, your only trying to deceive yourself 👍🏼 

Just a quote for you:
**_My unmatched perspicacity, coupled with sheer indefatigability, makes me a feared opponent in any realm of human endeavor_**

**Comment by MakcStudio** [comment]
What are you offended by? I work officially at the two largest p2p sites for trading skins. I've been in this industry for over 7 years. If you just googled my nickname, you would know that and not write nonsense. I explained why I didn't do the fork. Because I don't like English version, so I fixed mine, which I translated for my software years ago.

I just suggest you to make a fork and copy the solution from my repo. Unfortunately I do not have time to do the same job twice
